src/api/prop.js

```diff
@@ -26,9 +26,10 @@ export const boolean = create({
 });
 
 export const number = create({
-  coerce: alwaysUndefinedIfEmptyOrNumber,
-  deserialize: alwaysUndefinedIfEmptyOrNumber,
-  serialize: alwaysUndefinedIfEmptyOrNumber
+  default: 0,
+  coerce: alwaysZeroIfNotANumberOrNumber,
```

Comment:
Right. Hmm, I wonder what the expected behaviour here would be. `String('test')` really isn't a number.